yt/data_objects/data_containers.py

```diff
@@ -703,7 +703,7 @@ def to_glue(self, fields, label="yt", data_collection=None):
 
     def create_firefly_object(
         self,
-        JSONdir,
+        datadir,
```

Comment:
this particular change is not 100% backward compatible for yt users, we would also need to capture `JSONdir` being passed as a keyword argument and raise a deprecation warning when it's caugh (use `yt._maintainance.deprecation.issue_deprecation_warning`).
This also implies adding a default value to `datadir` (`None`)... well it gets messy pretty quickly. Let me know if you need more detailed instructions. I can also jump in and do this bit myself if you'd like.

Reply:
I think 12c54cc covers this? I followed the approach from yt/data_objects/image_array.py for the notation. I put in 4.2 as the removal version, but that was just a guess.

Reply:
Almost there. I'll add further comments.